### PR TITLE
add push command

### PR DESCRIPTION
**Pull request by nissimbd**
this is to push back our changes to the remote repository

### CODE REVIEW

File: etc/bash_completion.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 _git_vendor()
 {
-    __gitcomp "add list update"
+    __gitcomp "add list update push"
```
Comment (alzix) [inline etc/bash_completion.sh]: i think `upstream` better describes the operation

File: man/git-vendor.md [hunk 2]
```diff
@@ -11,13 +11,15 @@ git-vendor(1) -- manage vendored dependency subtrees
 
 `git-vendor update <name> [<ref>]`
 
+`git-vendor push <name> [<ref>]`
```
Comment (alzix) [inline man/git-vendor.md]: add new optional argument that can override `$repository` as in most cases you will not have write access to the origin repo and if you want to upstream your changes you would need to:
1. fork
2. upstream to the fork-repo
3. create pull- request from the fork-repo to the original repo

File: etc/bash_completion.sh (hunk 1 above)
Comment (nissimbd) [reply etc/bash_completion.sh]: thanks, I will

File: man/git-vendor.md (hunk 2 above)
Comment (nissimbd) [reply man/git-vendor.md]: for this I will need to parse the arguments and add the new optional argument with an option name to distinguish between ref and repo like:

git vendor upstream <name>  [**--repo** repository] and optional ref

File: bin/git-vendor [hunk 3]
```diff
@@ -207,6 +208,66 @@ git-vendor-ref: $ref
...
+
+            # And hasn't been renamed
+            logname=$(vendor_name_from_dir "$dir")
+            if [ "$name" != "$logname" ]; then
+                die "Dependency \"$1\" was renamed \"$logname\""
+            fi
+
+            if [ ! -z "$repository_arg" ];
+            then
+                # override the reposition read from the commit logs
```
Comment (brettlangdon) [inline bin/git-vendor]:
```suggestion
                # override the repository read from the commit logs
```

File: bin/git-vendor [hunk 4]
```diff
@@ -207,6 +208,66 @@ git-vendor-ref: $ref
...
+                # with the one read from the command line arguments
+                repository=$repository_arg
```
Comment (brettlangdon) [inline bin/git-vendor]:
```suggestion
                repository="$repository_arg"
```

File: man/git-vendor.md [hunk 5]
```diff
@@ -46,7 +52,7 @@ git-vendor(1) -- manage vendored dependency subtrees
 
   &lt;name&gt;
 
-  A name to provide the vendored dependency to use when listing/updating.
+  A name to provide the vendored dependency to use when listing/updating/pushing.
```
Comment (brettlangdon) [inline man/git-vendor.md:55]:
```suggestion
  A name to provide the vendored dependency to use when listing/updating/upstreaming.
```

Should it be "upstreaming" 🤔 , as an action "pushing" definitely sounds better here. Not sure I care enough about making my suggested change, just pointing out.

**Review by brettlangdon — COMMENTED** [review]
a few minor nits, but otherwise lgtm

File: man/git-vendor.md (hunk 5 above)
Comment (nissimbd) [reply man/git-vendor.md:55]: pushing sounds better for me too

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**